craft_application/services/remotebuild.py

```diff
@@ -289,21 +291,30 @@ def _ensure_recipe(
         repository: launchpad.models.GitRepository,
         **kwargs: Any,  # noqa: ANN401
     ) -> launchpad.models.Recipe:
-        """Create a new recipe."""
-        # There are 3 different ways this can fail in expected ways, all of them HTTPErrors:
-        #
-        # 1. Recipe doesn't exist (or you don't have access to see it)
-        # 2. You don't have access to delete the recipe
-        # 3. There is an ongoing build with the recipep
-        #
-        # In case 1, we don't care and are about to make the new recipe anyway.
-        # In the other cases, the errors from `new()` end up being more useful.
-        with contextlib.suppress(launchpadlib.errors.HTTPError):
+        """Get a recipe or create it if it doesn't exist."""
+
+        with contextlib.suppress(ValueError):  # Recipe doesn't exist
```

Comment:
Two PRs collided in a way that wasn't caught, updating the handling to a ValueError